src/main/java/com/synopsys/integration/detect/lifecycle/run/step/RapidModeStepRunner.java

```diff
@@ -40,16 +41,18 @@ public class RapidModeStepRunner {
     private final StepHelper stepHelper;
     private final Gson gson;
     private final DirectoryManager directoryManager;
+    private final DetectConfigurationFactory configurationFactory;
 
-    public RapidModeStepRunner(OperationRunner operationRunner, StepHelper stepHelper, Gson gson, DirectoryManager directoryManager) {
+    public RapidModeStepRunner(OperationRunner operationRunner, StepHelper stepHelper, Gson gson, DirectoryManager directoryManager, DetectConfigurationFactory configurationFactory) {
```

Comment:
Since configurationFactory.componentLocationAnalysisEnabled() alone is referred in this class, the boolean can be passed to this constructor.